actors/evm/src/interpreter/instructions/call.rs

```diff
@@ -196,29 +196,40 @@ pub fn call_generic<RT: Runtime>(
             &[]
         };
 
-        if precompiles::Precompiles::<RT>::is_precompile(&dst) {
+        if is_reserved_precompile_address(&dst.into()) {
             let context =
                 PrecompileContext { call_type: kind, gas_limit: effective_gas_limit(system, gas) };
-            if log::log_enabled!(log::Level::Info) {
-                // log input to the precompile, but make sure we dont log _too_ much.
-                let mut input_hex = hex::encode(input_data);
-                input_hex.truncate(512);
-                log::info!(target: "evm", "Calling Precompile:\n\taddress: {:x?}\n\tcontext: {:?}\n\tinput: {}", EthAddress::try_from(dst).unwrap_or(EthAddress([0xff; 20])), context, input_hex);
-            }
-
             match precompiles::Precompiles::call_precompile(system, dst, input_data, context) {
-                Ok(return_data) => (1, return_data),
-                Err(err) => {
-                    log::error!(target: "evm", "Precompile failed: error {:?}", err);
-                    // precompile failed, exit with reverted and no output
+                Some(res) => {
+                    if log::log_enabled!(log::Level::Info) {
+                        // log input to the precompile, but make sure we dont log _too_ much.
+                        let mut input_hex = hex::encode(input_data);
+                        input_hex.truncate(512);
+                        log::info!(target: "evm", "Call Precompile:\n\taddress: {:x?}\n\tcontext: {:?}\n\tinput: {}", EthAddress::try_from(dst).unwrap_or(EthAddress([0xff; 20])), context, input_hex);
+                    }
+
+                    match res {
+                        Ok(return_data) => (1, return_data),
+                        Err(err) => {
+                            log::error!(target: "evm", "Precompile failed: error {:?}", err);
+                            // precompile failed, exit with reverted and no output
+                            (0, vec![])
+                        }
+                    }
+                }
+                None => {
+                    log::warn!(target: "evm", "Non-existing precompile address: {:?}", EthAddress::from(dst));
                     (0, vec![])
```

Comment:
So, this should actually return success.

Reply:
Er, well... You know what? I don't care. Technically, we should return success.... but this is just so stupid.

Let's just return an error.